apple/REAModule.h

```diff
@@ -1,14 +1,24 @@
+#ifdef RCT_NEW_ARCH_ENABLED
+#import <rnreanimated/rnreanimated.h>
+#else
 #import <React/RCTBridgeModule.h>
+#endif
+#import <RNReanimated/REAAnimationsManager.h>
+#import <RNReanimated/REANodesManager.h>
 #import <React/RCTEventDispatcher.h>
 #import <React/RCTEventEmitter.h>
 #import <React/RCTUIManager.h>
 #import <React/RCTUIManagerObserverCoordinator.h>
 #import <React/RCTUIManagerUtils.h>
 
-#import <RNReanimated/REAAnimationsManager.h>
-#import <RNReanimated/REANodesManager.h>
-
-@interface REAModule : RCTEventEmitter <RCTBridgeModule, RCTEventDispatcherObserver, RCTUIManagerObserver>
+@interface REAModule : RCTEventEmitter
+#ifdef RCT_NEW_ARCH_ENABLED
+                       <NativeReanimatedModuleSpec,
+#else
+                       <RCTBridgeModule,
+#endif
+                        RCTEventDispatcherObserver,
+                        RCTUIManagerObserver>
```

Comment:
I'm wondering if this clever code is parsed correctly by Xcode intellisense; it certainly does not work well in GitHub syntax highlighting. How about the following?

```suggestion
#ifdef RCT_NEW_ARCH_ENABLED
typedef NativeReanimatedModuleSpec BaseModule;
#else
typedef RCTBridgeModule BaseModule;
#endif // RCT_NEW_ARCH_ENABLED

@interface REAModule : RCTEventEmitter <BaseModule, RCTEventDispatcherObserver, RCTUIManagerObserver>
```

Reply:
I works well in Xcode but I can change it.

Reply:
Hmm it looks like it has some problems with such syntax: 
![image](https://github.com/software-mansion/react-native-reanimated/assets/32481228/7b65602e-f6ef-4b78-bdfe-56fe00c53a5c)